src/pages/studyView/addChartButton/AddChartButton.tsx

```diff
@@ -428,6 +488,101 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
                             </div>
                         )}
                     </MSKTab>
+                    {!this.hideGenericAssayTabs &&
+                        _.map(
+                            this.props.store.genericAssayProfileOptionsByType
+                                .result,
+                            (options, type) => (
+                                <MSKTab
```

Comment:
this tab spec is too deeply nested.   lets make a wrapper component for it and extract some of the logic inside of attributes to local variables.

Reply:
Made a `@computed` component and separated logics into variables: https://github.com/cBioPortal/cbioportal-frontend/pull/3521/files#diff-2c76ee08342f5cf09e54594c2c9ea1b268689a000376f4b1e1e76d2f9fb048f5R232-R271